ast/src/main/java/com/graphicsfuzz/common/typing/TyperHelper.java

```diff
@@ -60,14 +60,16 @@ public static Type resolveTypeOfCommonBinary(Type lhsType, Type rhsType) {
 
     // If one side is scalar and the other side is basic, the result has to be that of the other
     // side
-    if (lhsType == BasicType.FLOAT || lhsType == BasicType.INT || lhsType == BasicType.UINT) {
-      assert rhsType instanceof BasicType;
-      return rhsType;
-    }
+    // If both are scalar but of different types (shift operator), the result type is the one
+    // from the left operand, so we check if the right operand is scalar first
```

Comment:
Can you clean up this comment a bit? The existing commentary potentially contradicts the new commentary. If one side is int and the other is uint, the first part of the commentary is ambiguous. And actually if the LHS is ivec2 but the RHS is uint, the result is ivec2 (in the case of a shift), contradicting the first part of the comment.

Feel free to simply rethink this whole section of comment to write something from scratch that reflects our new understanding.